File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count_map = {}

        for i in nums:
            if i in count_map:
                count_map[i] += 1
            else:
                count_map[i] = 1
        temp = [[] for _ in range(len(nums) + 1)]
        print(temp)
        for key, value in count_map.items():            
            temp[value].append(key)
            
        length = len(temp) - 1
        result = []
        while length >= 0 and k > 0:
            if temp[length]:
                result.extend(temp[length])
                k -= len(temp[length])
            length -= 1        
        
        return result

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        var previous = 1;
        var result = new int[nums.Length];
        result[0] = 1;

        for(int i = 1; i< nums.Length; i++)
        {
            previous *= nums[i - 1];
            result[i] = previous;
        }

        previous = 1;

        for(int i = nums.Length - 2; i >= 0; i--)
        {
            previous *= nums[i + 1];
            result[i] *= previous;
        }

        return result;
    }
}
